internal/provider/resource_hvn_peering_connection.go

```diff
@@ -0,0 +1,302 @@
+	}
+
+	peeringID := link.ID
+	loc := link.Location
+	hvnLink1, err := parseLinkURL(d.Get("hvn_1").(string), HvnResourceType)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	deletePeeringParams := network_service.NewDeletePeeringParams()
+	deletePeeringParams.Context = ctx
+	deletePeeringParams.ID = peeringID
+	deletePeeringParams.HvnID = hvnLink1.ID
+	deletePeeringParams.LocationOrganizationID = loc.OrganizationID
+	deletePeeringParams.LocationProjectID = loc.ProjectID
+
+	log.Printf("[INFO] Deleting network peering (%s)", peeringID)
+	deletePeeringResponse, err := client.Network.DeletePeering(deletePeeringParams, nil)
+	if err != nil {
+		if clients.IsResponseCodeNotFound(err) {
+			log.Printf("[WARN] Network peering (%s) not found, so no action was taken", peeringID)
+			return nil
+		}
+		return diag.Errorf("unable to delete network peering (%s): %v", peeringID, err)
+	}
+
+	// Wait for peering to be deleted
+	if err := clients.WaitForOperation(ctx, client, "delete network peering", loc, deletePeeringResponse.Payload.Operation.ID); err != nil {
+		if strings.Contains(err.Error(), "execution already started") {
+			return nil
+		}
+		return diag.Errorf("unable to delete network peering (%s): %v", peeringID, err)
+	}
+
+	log.Printf("[INFO] Network peering (%s) deleted, removing from state", peeringID)
+	return nil
+}
+
+func resourceHvnPeeringConnectionImport(ctx context.Context, d *schema.ResourceData, meta interface{}) ([]*schema.ResourceData, error) {
+	client := meta.(*clients.Client)
+	hvnID, peeringID, err := parsePeeringResourceID(d.Id())
+	if err != nil {
+		return nil, err
+	}
+
+	loc := &sharedmodels.HashicorpCloudLocationLocation{
+		ProjectID: client.Config.ProjectID,
+	}
+
+	link := newLink(loc, PeeringResourceType, peeringID)
+	url, err := linkURL(link)
+	if err != nil {
+		return nil, err
+	}
+
+	d.SetId(url)
+	if err := d.Set("hvn_id", hvnID); err != nil {
+		return nil, err
+	}
+
+	return []*schema.ResourceData{d}, nil
+}
+
+func setHvnPeeringResourceData(d *schema.ResourceData, peering *networkmodels.HashicorpCloudNetwork20200907Peering) error {
+	if err := d.Set("peering_id", peering.ID); err != nil {
+		return err
+	}
+	if err := d.Set("organization_id", peering.Hvn.Location.OrganizationID); err != nil {
+		return err
+	}
+
+	if err := d.Set("project_id", peering.Hvn.Location.ProjectID); err != nil {
+		return err
+	}
+	if err := d.Set("created_at", peering.CreatedAt.String()); err != nil {
+		return err
+	}
+	if err := d.Set("expires_at", peering.ExpiresAt.String()); err != nil {
+		return err
+	}
+
+	link := newLink(peering.Hvn.Location, PeeringResourceType, peering.ID)
+	selfLink, err := linkURL(link)
+	if err != nil {
+		return err
+	}
+	if err := d.Set("self_link", selfLink); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func getHvn(hvnSelfLink string, ctx context.Context, loc *sharedmodels.HashicorpCloudLocationLocation, client *clients.Client) (*networkmodels.HashicorpCloudNetwork20200907Network, error) {
```

Comment:
I wonder if this should be a general helper? Pretty sure a lot of our resources fetch HVNs 🤔